### ISSUE TITLE
bug: Compactor node panicked

### ISSUE THREAD

**Issue by fuyufjh** [risingwavelabs/risingwave#13676]
### Describe the bug

Compactor node panicked during longevity test

### Error message/log

```text
cannot convert 4333512855 into u32
stack backtrace:
   0:     0x55b28e1a81b7 - std::backtrace_rs::backtrace::libunwind::trace::he8dc9e634c3d5b69
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/std/src/../../backtrace/src/backtrace/libunwind.rs:93:5
   1:     0x55b28e1a81b7 - std::backtrace_rs::backtrace::trace_unsynchronized::hcc1fc094cc2c1843
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x55b28e1a81b7 - std::sys_common::backtrace::_print_fmt::he6c8e9059e21630d
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/std/src/sys_common/backtrace.rs:67:5
   3:     0x55b28e1a81b7 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h145361bce15f4a6a
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/std/src/sys_common/backtrace.rs:44:22
   4:     0x55b2858b4b50 - core::fmt::rt::Argument::fmt::hb890552c7d6e8c2b
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/core/src/fmt/rt.rs:142:9

   5:     0x55b2858b4b50 - core::fmt::write::h97d71643815ece8a
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/core/src/fmt/mod.rs:1117:17
   6:     0x55b28e1a0d5f - std::io::Write::write_fmt::h9145d12d2fb312aa
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/std/src/io/mod.rs:1762:15
   7:     0x55b28e1a7f94 - std::sys_common::backtrace::_print::h34d8c7ca46068513
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/std/src/sys_common/backtrace.rs:47:5
   8:     0x55b28e1a7f94 - std::sys_common::backtrace::print::h69de4357a88b4f9d
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/std/src/sys_common/backtrace.rs:34:9
   9:     0x55b28e1a9657 - std::panicking::default_hook::{{closure}}::hbf7d8b588847f0d6
  10:     0x55b28e1a93b5 - std::panicking::default_hook::h797e21049b5f2e1d
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/std/src/panicking.rs:292:9
  11:     0x55b28d5f1ff7 - <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call::hac595af44997dd58
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/alloc/src/boxed.rs:2021:9
  12:     0x55b28d5f1ff7 - risingwave_rt::panic_hook::set_panic_hook::{{closure}}::h7b8d23a8bed20d5d
                               at /risingwave/src/utils/runtime/src/panic_hook.rs:25:9
  13:     0x55b28d5f1ff7 - std::panicking::update_hook::{{closure}}::hf7890505dc5c798b
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/std/src/panicking.rs:234:47
  14:     0x55b28e1a9caa - <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call::h49df07f7cb25728d
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/alloc/src/boxed.rs:2021:9
  15:     0x55b28e1a9caa - std::panicking::rust_panic_with_hook::h4b1939882c095f9c
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/std/src/panicking.rs:735:13
  16:     0x55b28e1a9a2e - std::panicking::begin_panic_handler::{{closure}}::ha8983f1edca62350
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/std/src/panicking.rs:609:13
  17:     0x55b28e1a8676 - std::sys_common::backtrace::__rust_end_short_backtrace::h14cba478a9ee1fbd
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/std/src/sys_common/backtrace.rs:170:18
  18:     0x55b28e1a97a2 - rust_begin_unwind
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/std/src/panicking.rs:597:5
  19:     0x55b2858b1ab5 - core::panicking::panic_fmt::h6433adef86579fb6
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/core/src/panicking.rs:72:14
  20:     0x55b28d8af969 - risingwave_storage::hummock::sstable::utils::checked_into_u32::h1dd4544876bb5a12
                               at /risingwave/src/storage/src/hummock/sstable/utils.rs:164:19
  21:     0x55b28d8af969 - risingwave_storage::hummock::sstable::builder::SstableBuilder<W,F>::add_impl::{{closure}}::h08ff6a04a00956dc
                               at /risingwave/src/storage/src/hummock/sstable/builder.rs:345:25
  22:     0x55b28d8b9e70 - risingwave_storage::hummock::sstable::builder::SstableBuilder<W,F>::add::{{closure}}::hff75db046cde79da
                               at /risingwave/src/storage/src/hummock/sstable/builder.rs:284:46
  23:     0x55b28d8b9e70 - risingwave_storage::hummock::sstable::multi_builder::CapacitySplitTableBuilder<F>::add_full_key::{{closure}}::h1b04ac33e6830a37
                               at /risingwave/src/storage/src/hummock/sstable/multi_builder.rs:261:38
  24:     0x55b28d8b9e70 - risingwave_storage::hummock::compactor::fast_compactor_runner::CompactTaskExecutor<F>::run::{{closure}}::hee37ed622d8de94c
                               at /risingwave/src/storage/src/hummock/compactor/fast_compactor_runner.rs:638:18
  25:     0x55b28d8bcd68 - risingwave_storage::hummock::compactor::fast_compactor_runner::CompactorRunner::run::{{closure}}::hd93e7c0ba2ad880d
                               at /risingwave/src/storage/src/hummock/compactor/fast_compactor_runner.rs:436:49
  26:     0x55b28d8bcd68 - risingwave_storage::hummock::compactor::compactor_runner::compact::{{closure}}::{{closure}}::hb0d7a62cdfb7a476
                               at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/macros/select.rs:524:49
  27:     0x55b28d7b1663 - <tokio::future::poll_fn::PollFn<F> as core::future::future::Future>::poll::h67cf97cec6ee92f4
                               at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/future/poll_fn.rs:58:9
  28:     0x55b28d7b1663 - risingwave_storage::hummock::compactor::compactor_runner::compact::{{closure}}::hc39f727147383534
                               at /risingwave/src/storage/src/hummock/compactor/compactor_runner.rs:459:9
  29:     0x55b28d7c2d23 - risingwave_storage::hummock::compactor::start_compactor::{{closure}}::{{closure}}::h445e329a27e21410
                               at /risingwave/src/storage/src/hummock/compactor/mod.rs:514:185
  30:     0x55b28d79ff81 - <tracing::instrument::Instrumented<T> as core::future::future::Future>::poll::hce376fdf8f913d5a
                               at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tracing-0.1.37/src/instrument.rs:272:9
  31:     0x55b28d9cd47b - tokio::runtime::task::core::Core<T,S>::poll::{{closure}}::h78209f5697145a7a
                               at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/task/core.rs:334:17
  32:     0x55b28d9cd47b - tokio::loom::std::unsafe_cell::UnsafeCell<T>::with_mut::hda155873ce49a835
                               at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/loom/std/unsafe_cell.rs:16:9
  33:     0x55b28d9cd47b - tokio::runtime::task::core::Core<T,S>::poll::h5d39e04464cb7525
                               at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/task/core.rs:323:13
  34:     0x55b28da81b47 - tokio::runtime::task::harness::poll_future::{{closure}}::h8c2dad1f11e57f0b
                               at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/task/harness.rs:485:19
  35:     0x55b28da81b47 - <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once::heb8722ac95b7578b
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/core/src/panic/unwind_safe.rs:271:9
  36:     0x55b28da81b47 - std::panicking::try::do_call::h27b5c7ed6a148ec4
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/std/src/panicking.rs:504:40
  37:     0x55b28da81b47 - std::panicking::try::hfd7a92dcdf7845aa
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/std/src/panicking.rs:468:19
  38:     0x55b28da81b47 - std::panic::catch_unwind::h4d1e95e3f972813f
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/std/src/panic.rs:142:14
  39:     0x55b28da81b47 - tokio::runtime::task::harness::poll_future::hc63a818eaf10dbcf
                               at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/task/harness.rs:473:18
  40:     0x55b28da81b47 - tokio::runtime::task::harness::Harness<T,S>::poll_inner::ha40d2bd1059a4642
                               at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/task/harness.rs:208:27
  41:     0x55b28da81b47 - tokio::runtime::task::harness::Harness<T,S>::poll::hffe45a7e50f9e3cd
                               at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/task/harness.rs:153:15
  42:     0x55b28e228b2b - tokio::runtime::task::raw::RawTask::poll::h3411e2d2094a8398
                               at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/task/raw.rs:200:18
  43:     0x55b28e228b2b - tokio::runtime::task::LocalNotified<S>::run::h1b234fe523436f65
                               at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/task/mod.rs:400:9
  44:     0x55b28e228b2b - tokio::runtime::scheduler::multi_thread::worker::Context::run_task::{{closure}}::h43544b88b7324f96
                               at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/scheduler/multi_thread/worker.rs:639:22
  45:     0x55b28e228b2b - tokio::runtime::coop::with_budget::hc2e1ba91dc4343ba
                               at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/coop.rs:107:5
  46:     0x55b28e228b2b - tokio::runtime::coop::budget::h6914ce9705ab5185
                               at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/coop.rs:73:5
  47:     0x55b28e228b2b - tokio::runtime::scheduler::multi_thread::worker::Context::run_task::hf2150a2ed144755e
                               at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/scheduler/multi_thread/worker.rs:575:9
  48:     0x55b28e223264 - tokio::runtime::scheduler::multi_thread::worker::Context::run::h5a0f74052acdf7aa
                               at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/scheduler/multi_thread/worker.rs:526:24
  49:     0x55b28e223264 - tokio::runtime::scheduler::multi_thread::worker::run::{{closure}}::{{closure}}::h8e32ba4f9377cbeb
                               at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/scheduler/multi_thread/worker.rs:491:21
  50:     0x55b28e223264 - tokio::runtime::context::scoped::Scoped<T>::set::h50d293442ec3c30f
                               at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/context/scoped.rs:40:9
  51:     0x55b28e21f9ed - tokio::runtime::scheduler::multi_thread::worker::run::{{closure}}::h9ec63a91dd7464d8
                               at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/scheduler/multi_thread/worker.rs:486:9
  52:     0x55b28e21f9ed - tokio::runtime::context::runtime::enter_runtime::h79f151d8e5b7dcfe
                               at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/context/runtime.rs:65:16
  53:     0x55b28e22880c - tokio::runtime::scheduler::multi_thread::worker::run::h94f458d6e31ca12f
                               at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/scheduler/multi_thread/worker.rs:478:5
  54:     0x55b28e21fd83 - tokio::runtime::scheduler::multi_thread::worker::Launch::launch::{{closure}}::h8ad055d17f7134e5
                               at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/scheduler/multi_thread/worker.rs:447:45
  55:     0x55b28e21fd83 - <tokio::runtime::blocking::task::BlockingTask<T> as core::future::future::Future>::poll::h32b7aafb106ee064
                               at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/blocking/task.rs:42:21
  56:     0x55b28e23afab - <tracing::instrument::Instrumented<T> as core::future::future::Future>::poll::h3e192d8e607298d2
                               at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tracing-0.1.37/src/instrument.rs:272:9
  57:     0x55b28e254be1 - tokio::runtime::task::core::Core<T,S>::poll::{{closure}}::h5010035d46013761
                               at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/task/core.rs:334:17
  58:     0x55b28e254be1 - tokio::loom::std::unsafe_cell::UnsafeCell<T>::with_mut::h607b1db4fa129b9f
                               at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/loom/std/unsafe_cell.rs:16:9
  59:     0x55b28e254be1 - tokio::runtime::task::core::Core<T,S>::poll::ha745bcf27d80ea23
                               at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/task/core.rs:323:13
  60:     0x55b28e2359aa - tokio::runtime::task::harness::poll_future::{{closure}}::h63a6ace95d72ebe3
                               at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/task/harness.rs:485:19
  61:     0x55b28e2359aa - <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once::h3b36e779ade21bc0
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/core/src/panic/unwind_safe.rs:271:9
  62:     0x55b28e2359aa - std::panicking::try::do_call::hfc1abc43f6d847ad
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/std/src/panicking.rs:504:40
  63:     0x55b28e2359aa - std::panicking::try::hb1c8d83ed9254c57
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/std/src/panicking.rs:468:19
  64:     0x55b28e2359aa - std::panic::catch_unwind::h7f826221717df4ee
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/std/src/panic.rs:142:14
  65:     0x55b28e2359aa - tokio::runtime::task::harness::poll_future::hb9e09854f0339be2
                               at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/task/harness.rs:473:18
  66:     0x55b28e2359aa - tokio::runtime::task::harness::Harness<T,S>::poll_inner::ha3a343d2b9313af0
                               at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/task/harness.rs:208:27
  67:     0x55b28e2359aa - tokio::runtime::task::harness::Harness<T,S>::poll::h2e140b9d87d8ac8e
                               at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/task/harness.rs:153:15
  68:     0x55b28e249bde - tokio::runtime::task::raw::RawTask::poll::h3411e2d2094a8398
                               at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/task/raw.rs:200:18
  69:     0x55b28e249bde - tokio::runtime::task::UnownedTask<S>::run::h572704cc4b2af822
                               at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/task/mod.rs:437:9
  70:     0x55b28e249bde - tokio::runtime::blocking::pool::Task::run::h0685e9615187ae8e
                               at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/blocking/pool.rs:159:9
  71:     0x55b28e249bde - tokio::runtime::blocking::pool::Inner::run::h602aedddc3052c3c
                               at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/blocking/pool.rs:513:17
  72:     0x55b28e259410 - tokio::runtime::blocking::pool::Spawner::spawn_thread::{{closure}}::hc8d0833541aabfeb
                               at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/blocking/pool.rs:471:13
  73:     0x55b28e259410 - std::sys_common::backtrace::__rust_begin_short_backtrace::h5de4298adbaf4bd4
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/std/src/sys_common/backtrace.rs:154:18
  74:     0x55b28e24b29d - std::thread::Builder::spawn_unchecked_::{{closure}}::{{closure}}::h8faefb8db5ab8109
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/std/src/thread/mod.rs:529:17
  75:     0x55b28e24b29d - <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once::h6f640689fdeb9a50
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/core/src/panic/unwind_safe.rs:271:9
  76:     0x55b28e24b29d - std::panicking::try::do_call::h1cdc9b3cc1caa67b
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/std/src/panicking.rs:504:40
  77:     0x55b28e24b29d - std::panicking::try::h03c21a59e4dbd614
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/std/src/panicking.rs:468:19
  78:     0x55b28e24b29d - std::panic::catch_unwind::h41252732620c3f23
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/std/src/panic.rs:142:14
  79:     0x55b28e24b29d - std::thread::Builder::spawn_unchecked_::{{closure}}::hc4c4b0474cd10277
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/std/src/thread/mod.rs:528:30
  80:     0x55b28e24b29d - core::ops::function::FnOnce::call_once{{vtable.shim}}::h2fb3b88c7929fd95
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/core/src/ops/function.rs:250:5
  81:     0x55b28e1afe35 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::hac2e24a72c9237b1
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/alloc/src/boxed.rs:2007:9
  82:     0x55b28e1afe35 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::h0a4ea247e7a2feab
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/alloc/src/boxed.rs:2007:9
  83:     0x55b28e1afe35 - std::sys::unix::thread::Thread::new::thread_start::ha157bf15852ae2a4
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/std/src/sys/unix/thread.rs:108:17
  84:     0x7fba94e3bac3 - start_thread
                               at ./nptl/pthread_create.c:442:8
  85:     0x7fba94ecda40 - clone3
                               at ./misc/../sysdeps/unix/sysv/linux/x86_64/clone3.S:81
  86:                0x0 - <unknown>
```

### Additional context

https://grafana.test.risingwave-cloud.xyz/explore?panes=%7B%22doH%22:%7B%22datasource%22:%22PE59595AED52CF917%22,%22queries%22:%5B%7B%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22PE59595AED52CF917%22%7D,%22editorMode%22:%22code%22,%22expr%22:%22%7Bnamespace%3D%5C%22reglngvty-20231127-140235%5C%22,%20pod%3D~%5C%22benchmark-risingwave-compactor-c-b8f49898b-fdlt5%5C%22%7D%20%7C~%20%5C%22panic%5C%22%22,%22queryType%22:%22range%22,%22refId%22:%22A%22%7D%5D,%22range%22:%7B%22from%22:%221701094090000%22,%22to%22:%221701137423000%22%7D%7D%7D&schemaVersion=1&orgId=1

**Comment by Li0k** [comment]
I found the problematic task in the log

```
2023-11-28T01:45:18.004864681Z  INFO risingwave_storage::hummock::compactor::compactor_runner: Ready to handle compaction group 6 task: 235539 compact_task_statistics CompactTaskStatistics { total_file_count: 10, total_key_count: 8948874, total_file_size: 4424222254, total_uncompressed_file_size: 5144257707 } target_level 6 compression_algorithm 2 table_ids [1004] parallelism 1 task_memory_capacity_with_parallelism 30349174, enable fast runner: true
```
The task performs fast compact and the task size exceeds 4g, We suspect that a large number of block direct copies have occurred in the upper files. In the case of append blocks, there is a lack of judgment on the capacity of sst, because it cannot be clearly determined whether the user key is dispersed to different ssts, which leads to the occurrence of bugs.

In short, we need to check the capacity of the generating sst during the process of building sst using block copy in fast compact and degrade from block_copy to add_full_key in time. @Little-Wallace  is working on this fix

**Comment by Li0k** [comment]
fix pr https://github.com/risingwavelabs/risingwave/pull/13690
